tests/beta/queuing/test_beta_measure.py

```diff
@@ -250,3 +252,253 @@ def test_expand_no_observable(self):
 
         with pytest.raises(NotImplementedError, match="Cannot expand"):
             m.expand()
+
+
+class TestState:
+    """Tests for the state function"""
+
+    @pytest.mark.parametrize("wires", range(2, 5))
+    def test_state_shape_and_dtype(self, wires):
+        """Test that the state is of correct size and dtype for a trivial circuit"""
+
+        dev = qml.device("default.qubit", wires=wires)
+
+        @qnode(dev)
+        def func():
+            return state()
+
+        state_ev = func()
+        assert state_ev.shape == (1, 2 ** wires)
+        assert state_ev.dtype == np.complex128
+
+    def test_return_type_is_state(self):
+        """Test that the return type of the observable is State"""
+
+        dev = qml.device("default.qubit", wires=1)
+
+        @qnode(dev)
+        def func():
+            qml.Hadamard(0)
+            return state()
+
+        func()
+        obs = func.qtape.observables
+        assert len(obs) == 1
+        assert obs[0].return_type is State
+
+    @pytest.mark.parametrize("wires", range(2, 5))
+    def test_state_correct_ghz(self, wires):
+        """Test that the correct state is returned when the circuit prepares a GHZ state"""
+
+        dev = qml.device("default.qubit", wires=wires)
+
+        @qnode(dev)
+        def func():
+            qml.Hadamard(wires=0)
+            for i in range(wires - 1):
+                qml.CNOT(wires=[i, i + 1])
+            return state()
+
+        state_ev = func()[0]
+        assert np.allclose(np.sum(np.abs(state_ev) ** 2), 1)
+        assert np.allclose(state_ev[0], 1 / np.sqrt(2))
+        assert np.allclose(state_ev[-1], 1 / np.sqrt(2))
+
+    def test_return_with_other_types(self):
+        """Test that an exception is raised when a state is returned along with another return
+        type"""
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qnode(dev)
+        def func():
+            qml.Hadamard(wires=0)
+            return state(), expval(qml.PauliZ(1))
+
+        with pytest.raises(QuantumFunctionError, match="The state cannot be returned in combination"):
+            func()
+
+    @pytest.mark.parametrize("wires", range(2, 5))
+    def test_state_equal_to_dev_state(self, wires):
+        """Test that the returned state is equal to the one stored in dev.state for a template
+        circuit"""
+
+        dev = qml.device("default.qubit", wires=wires)
+
+        weights = qml.init.strong_ent_layers_uniform(3, wires)
+
+        @qnode(dev)
+        def func():
+            qml.templates.StronglyEntanglingLayers(weights, wires=range(wires))
+            return state()
+
+        state_ev = func()
+        assert np.allclose(state_ev, dev.state)
```

Comment:
Thought: some of these tests would be nice to be added to the shared device tests.

Reply:
Nice! I also want to add some tests to the shared suite that compare a given device to the results from `default.qubit` (we did this in `lightning.qubit`). Seems a good way to make sure everything is on a par. Will add to my list of ideas.